### PR TITLE
Default to https

### PR DESCRIPTION
**Pull request by Jason23347**
I'm not sure if I should "deep copy" `conn_t` instead of calling conn_set() in axel_* functions...

### CODE REVIEW

**Comment by ismaell** [comment]
Seems like this is not a good idea. I was hoping for it to not require any extra code, it would be acceptable if it requires changes to the interfaces or duplicates data at run time, but not if it adds more code.

**Comment by Jason23347** [comment]
Thank you for pointing out the problems with my code patiently，I will try to make changes mainly on `conn_set`.

**Comment by ismaell** [comment]
@Jason23347 I've closed the issue for now, as we'll not arrive at an acceptable solution at the moment. I think it would make more sense to restructure the way the URLs are queued first, then this feature will be more natural to implement.

**Comment by ismaell** [comment]
Right now the process of translation from input URL to internal state produces one target per input URL. In order to avoid changes in other parts of the code it should simply produce multiple candidates for URLs without scheme, one for HTTPS and another for HTTP.

The process then needs to be refined so that for example: secure protocols are tried first, and only if none works then we degrade to non-secure ones, which would be a generic improvement, complimentary, and self-contained.

Another possible improvement would be to group targets by hostname to avoid connecting to the same host via multiple protocols, because that's unlikely to improve the throughput.

In summary: we need to avoid special cases as much as possible.
